src/main/java/ch/njol/skript/SkriptConfig.java

```diff
@@ -332,111 +331,72 @@ private static void userDisableHooks(Class<? extends Hook<?>> hookClass, boolean
-				return false;
+				return;
 			}
-			
-			Config mc;
+
+			Config mainConfig;
 			try {
-				mc = new Config(configFile, false, false, ":");
-			} catch (final IOException e) {
-				Skript.error("Could not load the main config: " + e.getLocalizedMessage());
-				return false;
+				mainConfig = new Config(configFile, false, false, ":");
+			} catch (IOException ex) {
+				Skript.exception(ex, "Could not load the main config");
+				return;
 			}
-			mainConfig = mc;
+			SkriptConfig.mainConfig = mainConfig;
 
-			String configVersion = mc.get(version.key);
+			String configVersion = mainConfig.get(version.key);
 			if (configVersion == null || Skript.getVersion().compareTo(new Version(configVersion)) != 0) {
-				try {
-					final InputStream in = Skript.getInstance().getResource("config.sk");
-					if (in == null) {
+				if (!mainConfig.getMainNode().isValid()) {
+					Skript.error("Your config is outdated, but cannot be updated because it contains errors.");
+					return;
+				}
+
+				try (InputStream stream = Skript.getInstance().getResource("config.sk")) {
+					if (stream == null) {
 						Skript.error("Your config is outdated, but Skript couldn't find the newest config in its jar.");
-						return false;
-					}
-					final Config newConfig = new Config(in, "Skript.jar/config.sk", false, false, ":");
-					in.close();
-					
-					boolean forceUpdate = false;
-					
-					if (mc.getMainNode().get("database") != null) { // old database layout
-						forceUpdate = true;
-						try {
-							final SectionNode oldDB = (SectionNode) mc.getMainNode().get("database");
-							assert oldDB != null;
-							final SectionNode newDBs = (SectionNode) newConfig.getMainNode().get(databases.key);
-							assert newDBs != null;
-							final SectionNode newDB = (SectionNode) newDBs.get("database 1");
-							assert newDB != null;
-							
-							newDB.setValues(oldDB);
-							
-							// '.db' was dynamically added before
-							final String file = newDB.getValue("file");
-							assert file != null;
-							if (!file.endsWith(".db"))
-								newDB.set("file", file + ".db");
-							
-							final SectionNode def = (SectionNode) newDBs.get("default");
-							assert def != null;
-							def.set("backup interval", "" + mc.get("variables backup interval"));
-						} catch (final Exception e) {
-							Skript.error("An error occurred while trying to update the config's database section.");
-							Skript.error("You'll have to update the config yourself:");
-							Skript.error("Open the new config.sk as well as the created backup, and move the 'database' section from the backup to the start of the 'databases' section");
-							Skript.error("of the new config (i.e. the line 'databases:' should be directly above 'database:'), and add a tab in front of every line that you just copied.");
-							return false;
-						}
+						return;
 					}
-					
-					if (newConfig.setValues(mc, version.key, databases.key) || forceUpdate) { // new config is different
```

Comment:
I'm a little worried this behavior isn't maintained. Can you show the results of some tests with a config file updating from like, 2.4 -> 2.9.5 and likewise with 2.4->2.10? I'm not entirely sure how it worked tbh.